### DISCUSSION TITLE
How to rotate touchscreen input in sync with eDP output?

### DISCUSSION THREAD

**Discussion by XeCycle** [WayfireWM/wayfire#2477] [Q&A] [answered]
My device is ROG Flow X13, a 360-flip touchscreen laptop/tablet.  `wlr-randr` can successfully rotate eDP1, and I've confirmed iio-sensor-proxy can report current orientation; but touchscreen input doesn't rotate when I run `wlr-randr`, so I see inverted reactions to inputs.  How to do that without invoking libinput as root?

**Comment by soreau** [comment]
This sounds like a duplicate of [#1283](https://github.com/WayfireWM/wayfire/issues/1283#issuecomment-2283185138).

**Reply by XeCycle** [reply]
I failed to get that working, logs say I have
```
II 29-09-24 07:53:51.846 - [build/wayfire/src/core/seat/input-manager.cpp:47] handle new input: ELAN9008:00 04F3:40FD, default mapping: ELAN9008:00 04F3:40FD
II 29-09-24 07:53:51.846 - [build/wayfire/src/core/seat/input-manager.cpp:47] handle new input: ELAN9008:00 04F3:40FD Stylus, default mapping: ELAN9008:00 04F3:40FD Stylus
```

And I added config lines
```
[input-device:ELAN9008:00 04F3:40FD Stylus]
output = eDP-1

[input-device:ELAN9008:00 04F3:40FD]
output = eDP-1

[input-device:wlr_virtual_pointer_v1]
output = eDP-1
```

But it does not rotate.

**Reply by soreau** [reply]
Does it help if you use the [output] section in the config to set the transform?

**Reply by XeCycle** [reply]
Ah yes, it works if `output.transform` is set in config file.  How can I wire-up a dynamic behavior using data from iio-sensor-proxy, then?

**Reply by soreau** [reply]
Basically, it seems that some internal state isn't being updated for transforms and possibly other state, for the output. I can't say for sure, but does [this patch](https://termbin.com/k5kv) help?

**Comment by soreau** [comment, marked as answer]
Another option you might consider if you haven't enabled it already, is [this workaround](https://github.com/WayfireWM/wayfire/wiki/Configuration#workaroundsuse_external_output_configuration).

**Reply by XeCycle** [reply]
Ah yes, that fixes it when using `wlr-randr`.
